src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -77,6 +78,60 @@ struct ExeSpaceUtils {
   static TeamPolicy get_team_policy_force_team_size (Int ni, Int team_size) {
     return TeamPolicy(ni, team_size);
   }
+
+  template <typename TeamMember, class Lambda, typename ValueType>
+  static KOKKOS_INLINE_FUNCTION
+  void parallel_reduce (const TeamMember& team,
+			const int& begin,
+			const int& end,
+                        const Lambda& lambda,
+                        const bool serialize,
+                        ValueType& result)
+  {
+    if (serialize)
+    {
+      // We want to get C++ on GPU to match F90 on CPU. Thus, need to
+      // serialize parallel reductions.
+
+      // All threads init result.
+      // NOTE: we *need* an automatic temporary, since we do not know
+      //       where 'result' comes from. If result itself is an automatic
+      //       variable, using it would be fine (only one vector lane would
+      //       actually have a nonzero value after the single). But if
+      //       result is taken from a view, then all vector lanes would
+      //       see the updated value before the vector_reduce call,
+      //       which will cause the final answer to be multiplied by the
+      //       size of the warp.
+      auto local_tmp = Kokkos::reduction_identity<ValueType>::sum();
+
+      Kokkos::single(Kokkos::PerThread(team),[&] {
+      for (int k=begin; k<end; ++k)
+        lambda(k, local_tmp);
+        });
+
+      result = local_tmp;
+    }
+    else
+      Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+  }
+
+  template<typename PackType, typename ValueType, bool Serialize=false>
+  static KOKKOS_INLINE_FUNCTION
+  void reduce_add (const PackType& p,
```

Comment:
This should probably be in the ekat_pack.hpp file. I would also add it as a method of the pack class, actually, but free function is probably ok.

Reply:
If you don't mind, I'll implement there and then add the `column_reduce()` function here as discussed [here](https://github.com/E3SM-Project/EKAT/issues/26#issuecomment-688989928)

Reply:
It can go both ways (single PR vs two PRs). Multiple PRs make code review faster, but I don't feel strongly either way.

Reply:
I'll add it here